src/testing.c

```diff
@@ -108,16 +111,158 @@ void testing(){
     //Free input arrays
     free_8_16_64(input);
 
+    // Call 1.2. Comparison
+    //perf_test();
+}
 
+int perf_test(){
     //Create Random Numbers for performance test of array
     //8 | 32  | 64 | 128 | .... etc
-    printf("\n\n");printf(COLOR);printf("+-------------------------------------------+\n");
-    printf(COLOR);printf("| Randomized Arrays for performance testing |\n");
-    printf(COLOR);printf("+-------------------------------------------+\n");printf(RESET);
+    printf("\n\n");
+    printf(COLOR);printf("+--------------------------------+\n");
+    printf(COLOR);printf("| Arrays for performance testing |\n");
+    printf(COLOR);printf("+--------------------------------+\n");
+    printf(RESET);
+    printf("\nAll values in ticks\n\n");
+    
+    int **input_pt = (int **)malloc(7 * sizeof(int*)); 
+
+    if (randomizer_pt(input_pt) != 0) {
+        perror("ERROR: Problem in Random Number generation");
+        return 1;
+    }
 
-    int **input_pt = (int **)malloc(7 * sizeof(int*));
+    //show_randomizer_pt(input_pt, 20);
+    printf(COLOR);printf("Randomized\n");printf(RESET);
+    run_algos(input_pt);
+    printf("\n");
+    
+    // now arrays are sorted in ascending order, so are used for this test
+    printf(COLOR);printf("Ascending Order\n");printf(RESET);
+    run_algos(input_pt);
+    printf("\n");
+    
+    //sort array in descending order
+    int  exp = 1;
+    for (int i = 0; i < 7; i++) {
+        exp += 2;
+        reverse(input_pt, i, pow(2,exp));
+    }
+    
 
-    randomizer_pt(input_pt);
-    show_randomizer_pt(input_pt, 15);
+    // now arrays are sorted in descending order, so are used for this test
+    printf(COLOR);printf("Descending Order\n");printf(RESET);
+    run_algos(input_pt);
+    printf("\n");
+    
     free_pt(input_pt);
+    return 0;
+}
+
+void run_algos(int **begarr_pt) {
+    
+    printf("Array\t | BS |\t | IS |\t | MS |\t | QS |\n");
+
+    //define new pointer to pointer to int
+    int **endarr_pt = (int**)malloc(7 * sizeof(int)); 
+
+    for (int i = 0, exp = 1; i < 7; i++) {
+        exp += 2;
+        printf("%5i ", (int)pow(2,exp));
+        clock_t start = 0, end = 0;
+        //get memory for array 
+        *(endarr_pt+i) = (int*)malloc(pow(2,exp) * sizeof(int));   //do i need i here???
+       
+       
+        //take over (each time) "default" values for the current algo to run through array
+
+            
+            for(int j = 0; j < pow(2,exp);j++){
+                //*(endarr_pt[i] + j) = *(begarray_pt[i] + j); // do i have to have [i] here????
+                *(endarr_pt[i] + j) = *(begarr_pt[i] + j);
+            }
+
+        for(int j = 0; j < pow(2,exp);j++){
+            printf("%d ", *(endarr_pt[i] + j));
```

Comment:
Das printf kannst du ausschalten, damit er die Arrays nicht in der Konsole anzeigt - war nur als "CHECK" für mich gedacht, ob eh das richtige ankommt...